dev/loadgen/cmd/benchmark.go

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) 2020 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package cmd
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"syscall"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/protobuf/types/known/timestamppb"
+	"sigs.k8s.io/yaml"
+
+	"github.com/gitpod-io/gitpod/loadgen/pkg/loadgen"
+	"github.com/gitpod-io/gitpod/loadgen/pkg/observer"
+	"github.com/gitpod-io/gitpod/ws-manager/api"
+)
+
+var benchmarkOpts struct {
+	TLSPath string
+	Host    string
+}
+
+// benchmarkCommand represents the run command
+var benchmarkCommand = &cobra.Command{
```

Comment:
nit: Too big. Can you split this to call smaller functions?

Reply:
What's your level/criterion for an adequate function size?

Reply:
I find if functions can be refactored then it should be refactored to small functions. I don't think there could be/should be strict checks.

The primary reason I pointed this here is:
1. There are multiple variables being created in this function. The variables are complex and use other preceeding variables.
2. Variable/(s) have nested function in their field definition -> Hard to read and understand what is going on
3. The big constant string in the variable definition
4. When I try to review it becomes hard for me because now in step X the abstraction is less. I have to review not just the variable creation but also how it is being created.

Does it make any sense? From a reviewer perspective I find it hard to review. The code is good other wise.